client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -243,17 +291,44 @@ protected ParsedSchema getSchemaByIdFromRegistry(int id, String subject)
 
   private int getVersionFromRegistry(String subject, ParsedSchema schema, boolean normalize)
       throws IOException, RestClientException {
-    io.confluent.kafka.schemaregistry.client.rest.entities.Schema response =
-        restService.lookUpSubjectVersion(schema.canonicalString(),
-            schema.schemaType(), schema.references(), subject, normalize, true);
+    if (missingSchemaCache.getIfPresent(new SubjectAndSchema(subject, schema)) != null) {
+      throw new RestClientException("This ID is banned",
```

Comment:
This error message should be changed to be the same as the error message in the response from the server.